File: Programmers/Inho/Level01/문자열을 정수로 바꾸기.cs
using System;
public class Solution
{
    public int solution(string s)
    {
        int answer = 0;
        int i = 0;
        int temp = 0;
        while (i != (s.Length))
        {
            temp = 0;
            if (s[i] == '-')
            {
                for (int j = i + 1; j < (s.Length) && s[j] != '-' && s[j] != '+'; j++)
                {
                    temp -= (int)(s[j] - '0');
                    temp *= 10;
                    i++;
                }
                temp /= 10;
            }
            else
            {
                for (int j = i; j < (s.Length - 1) && s[j + 1] != '-' && s[j + 1] != '+'; j++)
                {
                    if (s[j] == '+')
                    {
                        i++;
                        continue;
                    }
                    temp += (int)(s[j] - '0');
                    temp *= 10;
                    i++;
                }
                temp += (s[i] - '0');
            }

            answer += temp;
            i++;
        }
        return answer;
    }
}

File: Programmers/Inho/C++/Level01/문자열을 정수로 바꾸기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(string s) {
    int answer = 0;
    // +이거나 없으면 양수를, -가 붙어있으면 음수를 반환
    // 그럼 음수만 조건걸고 나머지를 양수하면 될듯
    // s의 길이는 5이하라지만 그건 큰 상관 없을듯
    // 0도 고려
    return stoi(s);
}